vimeo-networking/src/main/java/com/vimeo/networking/model/error/InvalidParameter.java

```diff
@@ -54,16 +66,54 @@ public InvalidParameter() {}
 
     public InvalidParameter(String field, ErrorCode errorCode, String developerMessage) {
         this.mField = field;
-        this.mErrorCode = errorCode;
+        setErrorCode(errorCode);
         this.mDeveloperMessage = developerMessage;
     }
 
-    public String getField() {
-        return mField;
+    /**
+     * Set the raw error code, also sets the error code enum as a side effect.
+     *
+     * @param rawErrorCode the raw error code to set.
+     */
+    protected void setRawErrorCode(@Nullable String rawErrorCode) {
+        mRawErrorCode = rawErrorCode;
+        final ErrorCode errorCode = VimeoNetworkUtil.getGson().fromJson(mRawErrorCode, ErrorCode.class);
+        mErrorCode = errorCode != null ? errorCode : ErrorCode.DEFAULT;
+    }
+
+    /**
+     * Set the error code enum, also sets the raw error code string as a side effect.
+     *
+     * @param errorCode the error code enum to set.
+     */
+    private void setErrorCode(@NotNull ErrorCode errorCode) {
+        mErrorCode = errorCode;
+        final String json = VimeoNetworkUtil.getGson().toJson(errorCode);
+        mRawErrorCode = json.replaceAll("^\"|\"$", "");
     }
 
+    /**
+     * Returns the error code. If no value was set, defaults to {@link ErrorCode#DEFAULT}.
+     *
+     * @return the error code.
+     */
+    @NotNull
     public ErrorCode getErrorCode() {
-        return mErrorCode == null ? ErrorCode.DEFAULT : mErrorCode;
+        return mErrorCode;
+    }
+
+    /**
+     * Returns the raw error code, may be null.
+     *
+     * @return the raw error code as a nullable string.
+     */
+    @Nullable
+    public String getRawErrorCode() {
+        return mRawErrorCode;
+    }
+
+    public String getField() {
```

Comment:
This needs a java doc

Reply:
this method existed in the class prior to this PR, should I add javadocs to all methods in this class?

Reply:
Oh yeah, you just moved it... Obviously, it's better to have all of them if you have enough time to add them. I'll leave it up to you.